### PR TITLE
Remove production configuration that can be handled by HTTP(S) server

### PR DESCRIPTION
**Pull request by charn**
Since e.g. nginx is used to handle the HTTPS connection we
allow it to also control other security related headers like HSTS
and redirecting all traffic over to HTTPS.

### CODE REVIEW

File: config/settings/production.py [hunk 1]
```diff
@@ -33,14 +33,7 @@
 # Make sure SecurityMiddleware is listed first
 MIDDLEWARE_CLASSES = SECURITY_MIDDLEWARE + MIDDLEWARE_CLASSES
 
-# set this to 60 seconds and then to 518400 when you can prove it works
-SECURE_HSTS_SECONDS = env.int('DJANGO_SECURE_HSTS_SECONDS', default=60)
-SECURE_HSTS_INCLUDE_SUBDOMAINS = True
-SECURE_CONTENT_TYPE_NOSNIFF = True
-SECURE_BROWSER_XSS_FILTER = True
-SESSION_COOKIE_SECURE = True
```
Comment (jaywink) [inline config/settings/production.py]: I'm not sure what is the benefit from removing these from the Django settings. I'm not totally against it, but having them in the app configuration does at least force some good practises to people deploying the app. In cases where Django is able to affect something, after all it depends on the web server setup too.

However, `SESSION_COOKIE_SECURE` I'm pretty sure shouldn't be removed. It's not something afaict that can be set on web server side. If removal is to make sure http sites can be run - I don't think we should be encouraging that.

What is the main point for removing the settings?

Comment (charn) [reply config/settings/production.py]: Well true, `SESSION_COOKIE_SECURE` should probably not be removed, but moving other strictly header related out of Django allows them to be set and applied for all requests going through the web server.

Comment (charn) [reply config/settings/production.py]: Also added back `SECURE_SSL_REDIRECT`.

**Review by jaywink — APPROVED** [review]
I think this is better, time for mergy then! :shipit:

**Pull request merged**